starknet_py/net/account/account.py

```diff
@@ -290,6 +301,55 @@ async def get_nonce(
             self.address, block_hash=block_hash, block_number=block_number
         )
 
+    async def _check_outside_execution_nonce(
+        self,
+        nonce: int,
+        *,
+        block_hash: Optional[Union[Hash, Tag]] = None,
+        block_number: Optional[Union[int, Tag]] = None,
+    ) -> bool:
+        (is_valid,) = await self._client.call_contract(
+            call=Call(
+                to_addr=self.address,
+                selector=get_selector_from_name("is_valid_outside_execution_nonce"),
+                calldata=[nonce],
+            ),
+            block_hash=block_hash,
+            block_number=block_number,
+        )
+        return bool(is_valid)
+
+    async def get_outside_execution_nonce(self, retry_count=10) -> int:
+        while retry_count > 0:
+            random_stark_address = KeyPair.generate().public_key
+            if await self._check_outside_execution_nonce(random_stark_address):
+                return random_stark_address
+            retry_count -= 1
+        raise RuntimeError("Failed to generate a valid nonce")
```

Comment:
Is this logic (trying to get nonce in loop) described somewhere in [spec](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-9.md)? Could you please explain why such approach?

Reply:
That's a good question. I just (copied)[https://github.com/starknet-io/starknet.js/blob/6e353d3d50226907ce6b5ad53309d55ed51c6874/src/account/default.ts#L655] that approach from js library.

Reply:
Yeah, you are absolutely correct. Spec says: 
```
nonce: this is different from the account’s usual nonce, it is used to prevent signature reuse across executions and doesn’t need to be incremental as long as it’s unique.
```